### ISSUE TITLE
Bug: [no-useless-template-literals] Doesn't report for `String.raw`

### ISSUE THREAD

**Issue by StyleShit** [typescript-eslint/typescript-eslint#8989]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.4.3&fileType=.ts&code=MYewdgzgLgBNBOMC8MDKV4EswHMB08AhgO4AGAJAN4Dkh1AvldQEYOkDcQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS18iALbF4AQ2G1Gw6BPioMkRNGgdokcGAC%2BIHUA&tsconfig=&tokens=false

### Repro Code

```TypeScript
const str = String.raw`${'a'}${'b'}`;
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-useless-template-literals": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Report on the string

### Actual Result

The string is not reported

### Additional Info

Didn't know about this function. Discovered it [in this video](https://www.youtube.com/watch?v=SPaeA8shXFg) by @ThePrimeagen

____
Well, actually while writing this, I'm not sure how it'll be fixed in cases like this:

```TS
const str = String.raw`${ '\n' }---${ '\t' }`;
```

maybe we shouldn't?

**Comment by Josh-Cena** [comment]
I don't think this can be fixed in general. It requires us to have an understanding of what String.raw does and carving out special behaviors for it.

**Comment by kirkwaiblinger** [comment]
@StyleShit i had the same thought 😊, and I came to the same conclusion as @Josh-Cena

**Comment by bradzacher** [comment]
Yeah I'm in agreement - I think that this falls into the intersection of "niche usage" and "encoding api knowledge" - and I too think that it's okay that the rule doesn't have a special case for this.

3 votes against so closing.
